FlightAxis/flightaxis.py

```diff
@@ -264,6 +346,163 @@ def parseResponse(self, xml):
+        declination = field_value['declination']
+        self.mag_declination_deg = declination['value']
+        self.mag_declination_rad = math.radians(self.mag_declination_deg)
+        inclination = field_value['inclination']
+        self.mag_inclination_deg = inclination['value']
+        self.mag_inclination_rad = math.radians(self.mag_inclination_deg)
+        total_intensity = field_value['total-intensity']
+        self.mag_intensity_gauss = total_intensity['value']*FlightAxis.NANOTESLA_TO_GAUSS
+        print(f"Updated mag: mag_declination_deg={self.mag_declination_deg}, mag_inclination_deg={self.mag_inclination_deg}, mag_intensity_gauss={self.mag_intensity_gauss}")
+
+    
+    def getHilActuatorControls(self) -> bool:
+        """
+        Attempt to receive HIL_ACTUATOR_CONTROLS message and
+        save the actuator values
+
+        Returns True if the message was received
+
+        TODO: scale the controls properly
+        """
+        msg = self.connection.recv_match(blocking=False)
+        #msg = self.connection.recv_match(type='HIL_ACTUATOR_CONTROLS',blocking=True)
+        if msg:
+            #print(f"{msg}")
+            #for idx in range(1,FlightAxis.RC_CHANNLES):
+            #    self.rcin[idx] = msg.controls[idx]
+            return True
+        else:
+            return False
+
+        # t_start = time.time()
+        # msg = None
+        # while ( ((time.time() - t_start) < FlightAxis.TIMEOUT_S) or msg):
+        #     msg = self.connection.recv_match(type='HIL_ACTUATOR_CONTROLS',blocking=False)
+        #     #msg = self.connection.recv_match(blocking=False)
+        #     # type = 'HITL_RC_INPUTS_RAW' to check the RC inputs
+        # if msg:
+        #     print(f"{msg}")
+        #     for idx in range(1,FlightAxis.RC_CHANNLES):
+        #         self.rcin[idx] = msg.controls[idx]
+        #     return True
+        # else:
+        #     return False
+
+    def sendHilSensor(self):
+        """
+        Send HIL_SENSOR message
+        """
+        self.connection.mav.hil_sensor_send(
+            int(self.m_currentPhysicsTime_SEC * 1e6), # [usec] time
+            self.m_accelerationBodyAX_MPS2,# float [m/s/s] X acceleration
+            self.m_accelerationBodyAY_MPS2,# float [m/s/s] Y acceleration
+            self.m_accelerationBodyAZ_MPS2,# float [m/s/s] Z acceleration
+            # TODO: for now assume these are the same
+            math.radians(self.m_rollRate_DEGpSEC),# float [rad/s] Angular speed around X axis in body frame
+            math.radians(self.m_pitchRate_DEGpSEC),# float [rad/s] Angular speed around Y axis in body frame
+            math.radians(self.m_yawRate_DEGpSEC),# float [rad/s] Angular speed around Z axis in body frame
+            self.mag1.T.tolist()[0][0], # float [gauss] X Magnetic field
+            self.mag1.T.tolist()[0][1], # float [gauss] Y Magnetic field
+            self.mag1.T.tolist()[0][2], # float [gauss] Z Magnetic field
```

Comment:
variadics might be nice
```suggestion
            *self.mag1.T.tolist()[0], # float [gauss] X / Y / Z Magnetic field
```